library/core/src/str/pattern.rs

```diff
@@ -997,7 +998,7 @@ impl<'b> Pattern for &'b str {
 
                 #[cfg(all(target_arch = "x86_64", target_feature = "sse2"))]
                 if self.len() <= 32 {
-                    if let Some(result) = simd_contains(self, haystack) {
+                    if let Some(result) = simd_contains(self.as_bytes(), haystack.as_bytes()) {
```

Comment:
this function just takes a slice of bytes now. From what I can see the implementation does not rely on the input being UTF8 at all.